lib/streamlit/elements/widgets/number_input.py

```diff
@@ -263,14 +271,15 @@ def _number_input(
         args: WidgetArgs | None = None,
         kwargs: WidgetKwargs | None = None,
         *,  # keyword-only arguments:
+        placeholder: str | None = None,
         disabled: bool = False,
         label_visibility: LabelVisibility = "visible",
         ctx: ScriptRunContext | None = None,
     ) -> Number | None:
         key = to_key(key)
         check_callback_rules(self.dg, on_change)
         check_session_state_rules(
-            default_value=None if value is DefaultValue else value, key=key
+            default_value=value if value != "min" else None, key=key
```

Comment:
nit: we could have "min" as a constant defined in `widgets.py` similarly to how we have `DefaultValue` there.

Reply:
`min` in this case is a literal that is only specific/used for by number_input. `date_input` for example uses `today` and `time_input` uses `now` for the comparable purpose. `DefaultValue` on the other hand was shared across widgets. So, since it is widget-specific now, it's probably better to keep it within `number_input`.

Reply:
I will take a look later if I move it into a constant within number input

Reply:
Decided to keep it as is for now since with other comparable literal parameters in our API we are also just checking against the string, e.g. 

https://github.com/streamlit/streamlit/blob/764469105fa39ccd22c36572fa69551d23cebf58/lib/streamlit/elements/widgets/text_widgets.py#L302

https://github.com/streamlit/streamlit/blob/764469105fa39ccd22c36572fa69551d23cebf58/lib/streamlit/commands/page_config.py#L190